### PR TITLE
Feature: BeforeScopeInterface (2.next)

### PR DESCRIPTION
**Pull request by jamisonbryant**
Redo of #261 but with changes rebased on 2.next

### CODE REVIEW

**Comment by jamisonbryant** [comment]
OK, this time I _actually_ did it right and this branch is now based on `2.next` with _only_ my commits. 

It does still have the stan failure for `Identity.php` but there is another PR open to correct that.

**Comment by LordSimal** [comment]
The stan issue will be fixed by https://github.com/cakephp/authorization/pull/263 so we can ignore that.

**Review by LordSimal — APPROVED** [review]

**Comment by LordSimal** [comment]
@jamisonbryant all you have to do now is click on the triangle next to the `Update branch` button inside github, then select `Update with rebase` and click on `Rebase branch`

Then the CI in Github should be green.

**Comment by LordSimal** [comment]
I forgot that rebasing a fork has some additional steps. I'd say lets just merge this 😁 
But still I'd like @markstory comments on that.

File: src/Policy/BeforeScopeInterface.php [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+ * CakePHP(tm) : Rapid Development Framework (https://cakephp.org)
+ * Copyright (c) Cake Software Foundation, Inc. (https://cakefoundation.org)
+ *
+ * Licensed under The MIT License
+ * For full copyright and license information, please see the LICENSE.txt
+ * Redistributions of files must retain the above copyright notice.
+ *
+ * @copyright     Copyright (c) Cake Software Foundation, Inc. (https://cakefoundation.org)
+ * @link          https://cakephp.org CakePHP(tm) Project
+ * @since         3.0.0
```
Comment (ADmad) [inline src/Policy/BeforeScopeInterface.php]:
```suggestion
 * @since         2.4.0
```

**Review by ADmad — APPROVED** [review]

**Pull request merged**